security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -226,6 +231,26 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 	return ns, nil
 }
 
+// SecretExist checks if secret already existed.
+// This API is used for sds server to check if coming request is ack request.
+func (sc *SecretCache) SecretExist(connectionID, resourceName, token, version string) bool {
```

Comment:
This is not a new function and the logic is quite similar. Move this back to the original place? It will be much easier for code review.

Reply:
yes this PR just move this function around, prefer to move public APIs(like ```SecretExist``` to beginning of the file, and push helper function like ```keyCertRotationJob``` lower.